UniversalMediaEngine.Tests/MediaEngineTests.cs

```diff
@@ -0,0 +1,533 @@
+﻿using System;
+using Microsoft.VisualStudio.TestPlatform.UnitTestFramework;
+using System.Threading.Tasks;
+using Microsoft.Maker.Media.UniversalMediaEngine;
+using Windows.Storage;
+
+namespace UniversalMediaEngine.Tests
+{
+    [TestClass]
+    public class MediaEngineTests
+    {
+        [TestMethod]
+        public async Task TestInitializeSuceeds()
+        {
+            // Arrange
+            var engineUnderTest = new MediaEngine();
+
+            // Act
+            var result = await engineUnderTest.InitializeAsync();
+
+            // Assert
+            Assert.IsTrue(MediaEngineInitializationResult.Success == result);
+        }
+
+        [TestMethod]
+        public async Task TestPlayWithInitializeSucceeds()
+        {
+            // Arrange
+            var engineUnderTest = new MediaEngine();
+            bool callbackOccurred = false;
+            uint callbackTimeout = 10;
+
+            // Act
+            var result = await engineUnderTest.InitializeAsync();
+
+            engineUnderTest.MediaStateChanged += ((MediaState newState) => 
+            {
+                if (MediaState.Playing == newState)
+                    callbackOccurred = true;
+            });
+
+            Assert.IsTrue(MediaEngineInitializationResult.Success == result);
+
+            try
+            {
+                engineUnderTest.Play("ms-appx:///Assets/TestSilence.wav");
+
+                // Wait for the callback to be invoked until the timeout expires
+                uint counter = 0;
+                while (!callbackOccurred)
+                {
+                    if (counter == callbackTimeout)
+                        Assert.IsFalse(true, "Timed out waiting for callback from mediaengine");
```

Comment:
Might this be cleaner as:

Assert.IsFalse(counter == callbackTimeout, "timed out...")